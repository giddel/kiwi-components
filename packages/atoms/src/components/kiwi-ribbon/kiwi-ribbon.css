kiwi-ribbon {
  width: 85px;
  height: 88px;
  overflow: hidden;
  position: absolute;
  top: 0px;
  right: 0px;
  z-index: 10;
}

.ribbon-inner {
  text-align: center;
  transform: rotateZ(45deg);
  position: relative;
  padding: 7px 0;
  left: -5px;
  top: 11px;
  width: 120px;
  font-size: 15px;
  box-shadow: 0px -1px 2px 0px #000b;
}

.ribbon-inner:before,
.ribbon-inner:after {
  content: '';
  position: absolute;
}

.ribbon-inner:before {
  left: 0;
}

.ribbon-inner:after {
  right: 0;
}
