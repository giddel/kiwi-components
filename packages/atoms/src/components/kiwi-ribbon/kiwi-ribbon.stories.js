import { withKnobs, text, select } from '@storybook/addon-knobs';
import { html } from 'lit-html';
import notes from './readme.md';

export default {
  title: 'KiwiComponents/Ribbon',
  parameters: { notes },
  decorators: [withKnobs],
};

export const basic = () => {
  const backgroundColor = select(
    'backgroundColor',
    [
      'gray',
      'default',
      'primary',
      'secondary',
      'success',
      'info',
      'warning',
      'danger',
    ],
    'primary',
  );
  const textColor = select(
    'textColor',
    [
      'default',
      'primary',
      'secondary',
      'success',
      'danger',
      'warning',
      'info',
      'black',
      'paper',
      'gray',
      'charcoal',
    ],
    'default',
  );
  const ribbonText = text('text', 'Mockup');

  return html`<div class="panel panel-default m-1">
    <div class="panel-body" style="height: 150px;">
      <kiwi-ribbon
        text="${ribbonText}"
        background-color="${backgroundColor}"
        text-color="${textColor}"
      />
    </div>
  </div>`;
};
