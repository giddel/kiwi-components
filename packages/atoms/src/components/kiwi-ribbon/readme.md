# kiwi-ribbon



<!-- Auto Generated Below -->


## Properties

| Property          | Attribute          | Description                                                              | Type                                                                                                                                               | Default     |
| ----------------- | ------------------ | ------------------------------------------------------------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------- | ----------- |
| `backgroundColor` | `background-color` | The bootstrap background color scheme to use, e.g. primary, default, ... | `"danger" \| "default" \| "gray" \| "info" \| "primary" \| "secondary" \| "success" \| "warning" \| undefined`                                     | `'primary'` |
| `text`            | `text`             | The text to display in the ribbon                                        | `string \| undefined`                                                                                                                              | `undefined` |
| `textColor`       | `text-color`       | The bootstrap text color scheme to use, e.g. primary, default, ...       | `"black" \| "charcoal" \| "danger" \| "default" \| "gray" \| "info" \| "paper" \| "primary" \| "secondary" \| "success" \| "warning" \| undefined` | `'default'` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
