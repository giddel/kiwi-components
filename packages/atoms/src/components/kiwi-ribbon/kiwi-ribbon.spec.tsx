import { newSpecPage } from '@stencil/core/testing';
import { KiwiRibbon } from './kiwi-ribbon';

describe('kiwi-ribbon', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [KiwiRibbon],
      html: `<kiwi-ribbon></kiwi-ribbon>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
