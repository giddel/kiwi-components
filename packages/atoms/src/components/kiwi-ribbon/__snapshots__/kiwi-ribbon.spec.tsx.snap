// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kiwi-ribbon renders 1`] = `
<kiwi-ribbon>
  <div class="bg-primary ribbon-inner text-default"></div>
</kiwi-ribbon>
`;
